### PR TITLE
Improve test coverage related, ordering the entries to easy future deprecations.

### PR DESCRIPTION
**Pull request by svalqui**
This pull request addresses issue #1288 

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

reordered the deprecated file on a per release basis, older releases at he bottom, as is not ordered at the moment is hard to know where to put future deprecation (at the top or bottom).
this is to facilitate deprecation as we won't be improving test coverage for obsolete or deprecated modules.
alternatively it could be order alphabetically.

### CODE REVIEW

**Comment by peterjc** [comment]
The current order is intended to be Python releases, then modules/function; within that newest changes first.

Perhaps two top level headings would help clarify that split?

Within the code changes, multiple changes within a module make this a bit fuzzy.

You could make an argument for sorting the entries alphabetically? That would solve some sorting
difficulties.

**Comment by svalqui** [comment]
Updated the headers, Python releases first then code, most recent first. At the moment I'm not sure what is more practical if ordered by release number, most recent changes at the top, or alphabetically; so is ordered by release number, most recent changes on top.

File: DEPRECATED.rst [hunk 1]
```diff
@@ -1,123 +1,239 @@
...
+their functionality is now available through a read() function in
+``Bio.Motif.Parsers.AlignAce``.
+
+``MEMEParser``, ``_MEMEScanner``, ``_MEMEConsumer``, ``_MASTConsumer``,
+``MASTParser``, ``_MASTScanner``, and ``MASTRecord`` were declared obsolete in
+Release 1.54 and deprecated in Release 1.55 final; their functionality is now
+available through a ``read()`` function in ``Bio.Motif.Parsers.MEME`` and
+``Bio.Motif.Parsers.MAST``, respectively.
+
+``Bio.motifs.mast`` plain-text parsing deprecated in favor of XML parsing as of
```
Comment (peterjc) [inline DEPRECATED.rst]: This is about ``Bio.motifs`` not ``Bio.Motif``

**Comment by peterjc** [comment]
+1 on the section changes.

I can see three main uses of the file:

* users looking for a change (might know rough time, or just search for keyword)
* developers adding entries (new sections at top; but what if there is a relevant old entry?)
* developers looking for deprecated code ready to remove (checking top is easy; but have to search using release numbers in case change there is one lower down within an older module section)

Single module deprecations are easy.

The trouble is multiple rounds of function/class/etc deprecations within a module - and listing them all under a single section for that module. Do we sort it on the oldest deprecation (what we do now), or on the newest (moving the whole block to the top of the file as part of adding a new entry - which looks like what would be expected with the current PR content)?

Or, should we giving each deprecation its own section, which would make chronological sorting much easier?

**Comment by svalqui** [comment]
I think:
- If you are looking for a change better look at the history, as that also includes amendments, improvements not noted as deprecation notes; unclear whether sorted by release or alphabetical would be of help.
- Yes if you are adding new entries at the top is likely you ignore old entries further down, then is better to have it ordered alphabetically  (+1 Alphabetically ordered)
- Looking  for removing deprecated code, which is what I want to do because do not want to improve test coverage on those, I’m looking at it on a module basis, as they could be deprecated 3, 4 releases earlier; in this instance module name is more relevant than release number (1+ Alphabetically ordered)
- Multiple deprecation in a module, I’m thinking that if is ordered alphabetically by module, then the multiple entries/releases for the module are manageable, whether release number is top-bottom or bottom-up, is of less priority, leave it as is, older on top. (+1 Alphabetically ordered, with release order per module).
- If is order by release is easy to add new entries as you put them on top (+1 Release ordered)

So, I’m going to update this by alphabetical order; within the module, release ordered per module, older on top.

**Comment by peterjc** [comment]
@svalqui wait a little

@mdehoon you and I tend to do most of the routine removal of deprecated code, what are your thoughts in the ideal ordering of this file?

**Comment by mdehoon** [comment]
@peterjc  I have no preference.

**Comment by svalqui** [comment]
@peterjc any further thoughts? I would like to deprecated Bio.Crystal, and an obsolete method of Bio.Uniprot.

**Comment by peterjc** [comment]
Please open separate issues for those deprecation questions, cross reference #2223 for Bio.Crystal.

**Comment by peterjc** [comment]
The two levels of headers is an easy but useful change - let's start with that?

**Comment by svalqui** [comment]
Ok, this have the 2 level separation and the Bio.motifs comments are moved to its own header.

**Comment by peterjc** [comment]
Although not as dramatic a change as you initially suggested, this is an improvement - thank you!

**Pull request merged**